io.openems.common/src/io/openems/common/jsonrpc/response/QueryHistoricTimeseriesExportXlsxResponse.java

```diff
@@ -32,131 +38,246 @@
 	 * 
-	 * @param worksheet the WorkSheet
-	 * @param width     the width
-	 * @param columns   the number of columns
+	 * @param ws       the {@link Worksheet}
+	 * @param edgeId   the edgeId number
+	 * @param fromDate the fromdate the excel exported from
+	 * @param toDate   the todate the excel exported to
 	 */
-	public static void initializeColWidths(Worksheet worksheet, int width, int columns) {
-		for (int i = 0; i < columns; i++) {
-			worksheet.width(i, COL_WIDTH);
-		}
+	private static void addBasicInfo(Worksheet ws, String edgId, ZonedDateTime fromDate, ZonedDateTime toDate) {
+
+		DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MM/dd/yyyy");
+		String fromDateString = fromDate.format(formatter);
+		String toDateString = toDate.format(formatter);
+
+		addValueBold(ws, 0, 0, "FEMS-Nr.");
+		addValue(ws, 0, 1, edgId);
+		addValueBold(ws, 1, 0, "Export erstellt am");
+		addValue(ws, 1, 1, fromDateString);
+		addValueBold(ws, 2, 0, "Export Zeitraum");
+		addValue(ws, 2, 1, toDateString);
 	}
 
 	/**
-	 * Adds all data.
+	 * Adds the energy data header and values.
 	 * 
-	 * @param worksheet    the Worksheet
-	 * @param row          the start row
-	 * @param historicData the data map
+	 * @param ws   the {@link Worksheet}
+	 * @param data the energy data map
+	 * @throws OpenemsNamedException
 	 */
-	public static void addData(Worksheet worksheet, int row,
-			SortedMap<ZonedDateTime, SortedMap<ChannelAddress, JsonElement>> table) {
-		boolean isHeaderInitialized = false;
-		for (Entry<ZonedDateTime, SortedMap<ChannelAddress, JsonElement>> tableRow : table.entrySet()) {
-			if (!isHeaderInitialized) {
-				initializeDataHeader(worksheet, row++, tableRow.getValue());
-				isHeaderInitialized = true;
-			}
+	private static void addEnergyData(Worksheet ws, SortedMap<ChannelAddress, JsonElement> data)
+			throws OpenemsNamedException {
 
-			int col = 0;
-			worksheet.value(row, col++, tableRow.getKey().toString());
-			for (Entry<ChannelAddress, JsonElement> tableColumn : tableRow.getValue().entrySet()) {
-				JsonElement jValue = tableColumn.getValue();
-				if (jValue.isJsonNull()) {
-					continue;
-				}
-				double value = jValue.getAsDouble();
-				worksheet.value(row, col++, Math.round(value));
+		// Heading for Grid buy energy
+		addValueBold(ws, 4, 1, "Netzbezug [kWh]");
+
+		// Grid buy energy
+		if (!data.get(GRID_BUY_ENERGY).isJsonNull()) {
+			addValue(ws, 5, 1, JsonUtils.getAsInt(data.get(GRID_BUY_ENERGY)));
+		} else {
+			addValue(ws, 5, 1, "NA");
+		}
+		// Heading for Grid sell energy
+		addValueBold(ws, 4, 2, "Netzeinspeisung [kWh]");
+
+		// Grid sell energy
+		if (!data.get(GRID_SELL_ENERGY).isJsonNull()) {
+			addValue(ws, 5, 2, JsonUtils.getAsInt(data.get(GRID_SELL_ENERGY)));
+		} else {
+			addValue(ws, 5, 2, "NA");
+		}
+
+		// Heading for Production energy
+		addValueBold(ws, 4, 3, "Erzeugung [kWh]");
+
+		// Production energy
+		if (!data.get(PRODUCTION_ACTIVE_ENERGY).isJsonNull()) {
```

Comment:
This `if` for JsonNull is happening very often. Can you instead create a smarter variant of `addValue` method that is able to do this automatically? Also "NA" will not be understood by any (German) user. Better would the more common "N/A" (https://en.wikipedia.org/wiki/N/a, https://de.wikipedia.org/wiki/N/A) or even "nicht verfügbar" to be very clear. It could also be formatted differently in that case - e.g. italic